llvm/lib/IR/IntrinsicInst.cpp

```diff
@@ -248,6 +250,9 @@ int llvm::Intrinsic::lookupLLVMIntrinsicByName(ArrayRef<const char *> NameTable,
   // identical. By using strncmp we consider names with differing suffixes to
   // be part of the equal range.
   size_t CmpEnd = 4; // Skip the "llvm" component.
+  if (!Target.empty())
+    CmpEnd += 1 + Target.size(); // skip the .target component.
```

Comment:
```suggestion
    CmpEnd += Target.size() +1; // skip the .target component.
```

Reply:
The 1 + size() to logically match ".target".